ext/mysqli/mysqli_api.c

```diff
@@ -729,14 +731,21 @@ PHP_FUNCTION(mysqli_data_seek)
 		RETURN_THROWS();
 	}
 
+	if (offset < 0) {
+		zend_argument_value_error(ERROR_ARG_POS(2), "must be greater than or equal to 0");
+		RETURN_THROWS();
+	}
+
 	MYSQLI_FETCH_RESOURCE(result, MYSQL_RES *, mysql_result, "mysqli_result", MYSQLI_STATUS_VALID);
 
 	if (mysqli_result_is_unbuffered(result)) {
+		// TODO Promoto to Exception?
 		php_error_docref(NULL, E_WARNING, "Function cannot be used with MYSQL_USE_RESULT");
 		RETURN_FALSE;
 	}
 
-	if (offset < 0 || (uint64_t)offset >= mysql_num_rows(result)) {
+	if ((uint64_t)offset >= mysql_num_rows(result)) {
```

Comment:
I'm not sure about it. It would require to call `mysqli_num_rows()` all the time if programmers want to avoid a possible exception. Or is it ok performance-wise?

Reply:
It's better to have an exception when you go out of bounds rather than having unexpected behaviour.

Reply:
There is no unexpected behaviour because it returns false and it is a pre-check. But those are more run-times errors compared to programming error so I'm not sure either if I should go with it

Reply:
As much as I don't want to agree, I have to admit you are right. Your arguments make sense.
However, the other two errors ("Function cannot be used in MYSQLI_USE_RESULT mode") I suggested in my PR still make sense in my opinion. Should I withdraw my PR and submit a new one?

Reply:
You can just amend your PR :)